<launch>

<!-- Incluimos el launch para ejecutar el mundo de gazebo -->
<!--<include file="$(find mybot_gazebo)/launch/mybot_world.launch"/>-->

<include file="$(find turtlebot_navigation)/launch/includes/move_base_rtab.launch.xml"/>

<param name="use_sim_time" type="bool" value="True"/>

<!--delete_db_on_start como argumento-->
<group ns="rtabmap">   
<node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="">

         <param name="frame_id" type="string" value="pioneer2dx/chassis"/>
	 <param name="odom_frame_id" type="string" value="odom"/>
	<param name="map_frame_id" type="string" value="map"/>
         
         <param name="subscribe_depth" type="bool" value="true"/>
	
         

         <remap from="odom" to="/odom"/>
	<remap from="grid_map" to ="/map"/>
	<remap from="depth/image" to ="/camera_ir/depth/image_raw"/>
	<remap from="rgb/image" to ="/camera_ir/rgb/image_raw"/>
	<remap from="rgb/camera_info" to ="/camera_ir/rgb/camera_info"/>

         <param name="queue_size" type="int" value="30"/>

         <!-- RTAB-Map's parameters -->
	<!--En caso de querer poner el modo localización debemos poner el parámetro Mem/IncrementalMemory a falso y Mem/InitWMWithAllNodes a true-->
         
	<param name="Rtabmap/TimeThr" type="string" value="0"/>
         <param name="Rtabmap/DetectionRate" type="string" value="0"/>
         
         <param name="Kp/WordsPerImage" type="string" value="-1"/>
         <param name="Kp/DetectorStrategy" type="string" value="0"/>   <!-- use SURF -->
         <param name="Kp/NNStrategy" type="string" value="1"/>         <!-- kdTree -->
	 <param name="Grid/CellSize" type="string" value="0.01"/> <!-- 1 cm voxel -->
         <param name="SURF/HessianThreshold" type="string" value="1000"/>

         <param name="Vis/MinInliers" type="string" value="10"/>
         <param name="Vis/EstimationType" type="string" value="1"/> <!-- 3D->2D (PnP) -->

         <param name="RGBD/LoopClosureReextractFeatures" type="string" value="true"/>
	<!--Number of features used for odometry (rgbd_odometry node): Vis/MaxFeatures (default 1000) -->         
	<param name="Vis/MaxFeatures" type="string" value="500"/> 
	
         <param name="Vis/MaxDepth" type="string" value="10"/> 

	 
      </node>
</group>
<!--
<param name="subscribe_stereo" type="bool" value="true"/>
<remap from="left/image_rect" to="/multisense_sl/camera/left/image_rect_color"/>
 <remap from="right/image_rect" to="/multisense_sl/camera/right/image_rect_color"/>
 <remap from="left/camera_info" to="/multisense_sl/camera/left/camera_info"/>
 <remap from="right/camera_info" to="/multisense_sl/camera/right/camera_info"/>

-->

</launch>
